### PR TITLE
Adds major version 10 featuring WASM

### PR DESCRIPTION
**Pull request by scottopell**
This refactors all existing code into v5 (FFI) and adds a new v10 (WASM).

Currently this also does not add the pre-built wasm binary blobs, that will be done as part of #1 

This has a few minor cleanup TODOs left, and a few VRL features left to support to reach parity with V5 in terms of features.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,15 +5,51 @@ Experimental go bindings for [Vector Remap Language](https://vector.dev/docs/ref
...
+  some applications may not care for.
+- **V10** uses `wasm` to execute VRL. It performs worse, on the order of 2-3 times
+  slower, however VRL is quite efficient so this still offers relatively good
+  absolute performance.
+
 ## Usage
 
+### Feature Support
+
+|                           | V5 | V10 |
```
Comment (gh123man) [inline README.md:26]: I think `V5` and `V10` are flipped here as `V5` is the CGO variant and supports `Kind`?

File: README.md [hunk 2]
```diff
@@ -5,15 +5,51 @@ Experimental go bindings for [Vector Remap Language](https://vector.dev/docs/ref
...
+|-------------------------- | -- | --- |
+| Compiling a VRL Program   | ✅ | ✅  |
+| Running a VRL Program     | ✅ | ✅  |
+| VRL Runtime "Basic"\* API     | ✅ | ✅  |
+| Environment Kinds     | ❌ | 'Byte' and 'Object'  |
+| Secrets                   | ❌ | ❌  |
+| Metadata                  | ❌ | ❌  |
+| Timezones                 | ❌ | ❌  |
+| Requires CGO              | ❌ | ✅  |
```
Comment (gh123man) [inline README.md:35]: too big to fit in this table - but IIRC, the WASM version has a limited exposure to the VRL std lib right (some functions dont work)? May be worth a note.

File: README.md [hunk 3]
```diff
@@ -5,15 +5,51 @@ Experimental go bindings for [Vector Remap Language](https://vector.dev/docs/ref
...
+
+\* "Basic" API currently means:
```
Comment (gh123man) [inline README.md:37]: + `- `compile`

File: v10/wasminterface.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	"fmt"
+	"log"
+	"runtime"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+)
+
+//go:embed target/wasm32-wasi/release/vrl_bridge.wasm
```
Comment (gh123man) [inline v10/wasminterface.go:15]: this is cool

File: v10/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+    // into_raw leaks the memory to the caller.
+    Box::into_raw(vec.into_boxed_slice()) as *mut u8
+}
+
+/// WebAssembly export that deallocates a pointer of the given size (linear
+/// memory offset, byteCount) allocated by [`allocate`].
+#[cfg_attr(all(target_arch = "wasm32"), export_name = "deallocate")]
+#[no_mangle]
+pub unsafe extern "C" fn _deallocate(ptr: u32, size: u32) {
```
Comment (gh123man) [inline v10/src/lib.rs:114]: is this unused?

**Review by gh123man — APPROVED** [review]
🚀
Just some minor comments. I trust your WASM integration as I don't know in great detail how that all works. But if the tests work 👍 and im sure anything else will be caught with more development as m(if?) we address the TODOs

File: README.md (hunk 2 above)
Comment (scottopell) [reply README.md:35]: yes, good call, added.

File: README.md (hunk 1 above)
Comment (scottopell) [reply README.md:26]: Definitely, good catch.

**Pull request merged**